app/lib/import_export.rb

```diff
@@ -541,25 +231,67 @@ def get_export_data(patients, data)
     exported_data
   end
 
-  # Extracts patient data values given relevant fields
+  # Extract patient data values given relevant fields
   def extract_patients_details(patients, fields)
-    # perform the following queries in bulk only if requested for better performance
-    if fields.include?(:jurisdiction_name)
-      jurisdiction_names = Hash[Jurisdiction.find(patients.pluck(:jurisdiction_id).uniq).pluck(:id, :name).map { |id, name| [id, name] }]
-    end
-    if fields.include?(:jurisdiction_path)
-      jurisdiction_paths = Hash[Jurisdiction.find(patients.pluck(:jurisdiction_id).uniq).pluck(:id, :path).map { |id, path| [id, path] }]
+    # query jurisdiction paths in bulk if requested
+    if (fields & %i[jurisdiction_path transferred_from transferred_to]).any?
+      jur_ids = patients.pluck(:jurisdiction_id).uniq
+      jur_ids = (jur_ids + patients.pluck(:latest_transfer_from)).uniq if fields.include?(:transferred_from)
+      jurisdiction_paths = Hash[Jurisdiction.find(jur_ids).pluck(:id, :path)]
     end
 
-    patients_transfers = get_patients_transfers(patients) if (fields & %i[transferred_from transferred_to]).any?
-    patients_laboratories = get_patients_laboratories(patients) if (fields & PATIENT_LAB_FIELDS).any?
-    patients_creators = Hash[User.find(patients.pluck(:creator_id)).pluck(:id, :email)] if fields.include?(:creator)
+    # query jurisdiction names in bulk if requested
+    jurisdiction_names = Hash[Jurisdiction.find(patients.pluck(:jurisdiction_id).uniq).pluck(:id, :name)] if fields.include?(:jurisdiction_name)
+
+    # query user emails in bulk if requested
+    patients_creators = Hash[patients.joins('INNER JOIN users ON patients.creator_id = users.id').pluck('users.id', 'users.email')] if fields.include?(:creator)
```

Comment:
I'm using raw SQL here because ActiveRecord doesn't know how to join patients to users by just writing: `patients.joins(:users)` since the foreign key is `creator_id` instead of `user_id`. Maybe there's a way to set that up in the model somewhere, but that's the way it's currently setup